transport.go

```diff
@@ -90,39 +91,40 @@ func (t *transport) IsConnected() bool {
 	}
 }
 
-func (t *transport) Run() error {
-	if !t.IsConnected() {
```

Comment:
Well if you've gotten rid of the error chan, then you don't need to do anything with the error.